datasource/opennms-common/src/main/java/org/opennms/oce/datasource/common/inventory/script/AbstractScriptedInventory.java

```diff
@@ -0,0 +1,199 @@
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.oce.datasource.common.inventory.script;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+import javax.script.Invocable;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Files;
+
+/**
+ * @author smith
+ *
+ */
+public abstract class AbstractScriptedInventory {
+
+    protected static final Logger LOG = LoggerFactory.getLogger(AbstractScriptedInventory.class);
+
+    protected static final String DEFAULT_SCRIPT = "/inventory.groovy";
+
+    private boolean usingClasspathScript;
+
+    private BundleContext context;
+
+    private Invocable invocable;
+
+    private String scriptPath;
+
+    private ScriptEngineManager manager;
+
+    private ScriptEngine engine;
+
+    private long configurationTimestamp;
+
+    private long scriptFileTimestamp;
+
+    private long scriptCacheMillis;
+
+    public AbstractScriptedInventory(String scriptPath, long scriptCacheMillis, BundleContext bundleContext) {
+        if (scriptPath == null) {
+            throw new IllegalArgumentException("Null value for scriptFile.");
+        }
+
+        String script;
+        String scriptExtension;
+
+        this.scriptCacheMillis = scriptCacheMillis;
+
+        if (scriptPath.isEmpty()) {
+            // load default from classpath
+            usingClasspathScript = true;
+            URL scriptUrl = bundleContext.getBundle().getResource(DEFAULT_SCRIPT);
+            StringBuilder sb = new StringBuilder();
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(scriptUrl.openStream()))) {
```

Comment:
Can be done more concisely using `CharStreams.toString` from Guava

Reply:
done